docs/asyncapi-file.md

```diff
@@ -3,4 +3,89 @@ title: "AsyncAPI Specification File"
+
+In the following sections, you'll learn about the inner working of the generator, what happens once the specification file is fed to the generator, and how you can use the content of the specification file with either an AsyncAPI document or AsyncAPI string in your template.
+
+### The generation process
+1. The **generator library** receives **[AsyncAPI specification file](asyncapi-file.md)** as an input. 
+2. To validate the json/yaml specification file, the generator passes the stringified version of the original specification document to the **[parser](parser.md)**. 
+3. The parser validates the AsyncAPI specification using either the OpenAPI, RAML, or Avro schema defined in the specification files. 
+4. If valid, the parser then manipulates the original JSON or YML specification file into functions and properties, bundling them together into an **asyncapiDocument**. 
+5. At this point, the generator library passes the **original asyncAPIString** and the the **asyncAPIDocument** which make up part of the [template context](asyncapi-context.md). 
+6. The generator passes the **template context** to either the [react](react-render-engine.md) or [nunjucks](nunjucks-render-engine.md)render engine therefore making it accessible in the templates.
+   
+``` mermaid
+graph LR
+    A[Template Context]
+    B{Generator}
+    C[Parser]
+    D[Render Engine]
+    E[AsyncAPI File] --> B
+  subgraph Generator Library
+    B -->| asyncapiString | C
+    C --> | asyncapi -> AsyncAPIDocument type | A
+    B--> | originalAsyncAPI -> Stringified document | A
+    A --> D
+  end
+  ```
+
+> :bulb: **Remember**
+> 
+> When you are writing your template, you get access to the specification file in two different forms; the originalAsyncAPI which is a stringified version of the specification `.yml` file, and the asyncapiDocument.
+
+> :memo: **Note:** 
+> Check out the [asyncapiDocument structure](https://github.com/asyncapi/parser-js/blob/master/API.md#module_@asyncapi/parser+AsyncAPIDocument)
+
+In the following sections, you will learn how to use either the **asyncapiString** or the **asyncapiDocument** in your template.
+
+#### Method 1: asyncapiString + template 
+The first way to generate a message-based API template using the generator tool is by using the asyncapiString. The asynapiString is a stringified version of the specification YAML or JSON file you pass as a parameter to the CLI `ag` command. During the generation process, the generator library calls `generator.originalAsyncAPI`, which returns a stringified version of the original spec file. This stringified version is what the generator library passes to the render engine, and the render engine makes it accessible in your output alias the template.
```

Comment:
```suggestion
One way of using the contents of AsyncAPI file inside your template files is by using its stringified version that reflects exactly the same contents as the file provided as an input. You can access this stringified AsyncAPI file directly in your templates through `originalAsyncAPI` variable. You also have access to it in hooks functionality because `originalAsyncAPI` is also a part of the Generator instance that is passed to hooks so that you can access it like `generator.originalAsyncAPI`.
```

Reply:
**Note to self** Link hooks.md here